keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -293,33 +291,49 @@ def _augment(self, inputs):
         return result
 
     def _batch_augment(self, inputs):
-        return self._map_fn(self._augment, inputs)
+        augmented_batch = self._map_fn(self._augment, inputs)
+        return augmented_batch
 
     def _format_inputs(self, inputs):
         if tf.is_tensor(inputs):
             # single image input tensor
-            return {IMAGES: inputs}, False, False
+            meta_data = {"is_dict": False, "use_targets": False}
+            inputs = {IMAGES: inputs}
         elif isinstance(inputs, dict) and TARGETS in inputs:
             # TODO(scottzhu): Check if it only contains the valid keys
             inputs[LABELS] = inputs[TARGETS]
             del inputs[TARGETS]
-            return inputs, True, True
+            meta_data = {"is_dict": True, "use_targets": True}
         elif isinstance(inputs, dict):
-            return inputs, True, False
+            meta_data = {"is_dict": True, "use_targets": False}
         else:
             raise ValueError(
                 f"Expect the inputs to be image tensor or dict. Got {inputs}"
             )
+        if BOUNDING_BOXES in inputs:
+            if isinstance(inputs[BOUNDING_BOXES], tf.RaggedTensor):
+                meta_data["is_bounding_boxes_ragged"] = True
+                meta_data["ragged_row_lengths"] = inputs[
+                    BOUNDING_BOXES
+                ].nested_row_lengths()
+                inputs[BOUNDING_BOXES] = inputs[BOUNDING_BOXES].to_tensor()
+            else:
+                meta_data["is_bounding_boxes_ragged"] = False
+        return inputs, meta_data
 
-    def _format_output(self, output, is_dict, use_targets):
-        if not is_dict:
+    def _format_output(self, output, meta_data):
+        if not meta_data["is_dict"]:
             return output[IMAGES]
-        elif use_targets:
+        elif meta_data["use_targets"]:
```

Comment:
lets split these checks for readability:

```
if not metadata["is_dict"]:
...

if metadata["use_targets"]:
   ....

return output
```

When using a guard clause (an early if with a return statement) it is easier to mentally follow if the following statements are followed by a newline and a new set of ifs

Reply:
done